src/emerge.cpp

```diff
@@ -737,6 +737,8 @@ void *EmergeThread::run()
 
 			if (!error)
 				block = finishGen(pos, &bmdata, &modified_blocks);
+			else
+				m_map->cancelBlockMake(&bmdata);
```

Comment:
What happens in the `!block` case? Do we need to cancel the blockMake in that case as well?